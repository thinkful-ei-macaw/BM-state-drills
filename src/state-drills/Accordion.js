import React from 'react';

class Accordion extends React.Component {
    static defaultProps = { sections: [] };
    constructor(props) {
    super(props)
    this.state = {
        currentSectionIndex: null,
        
    }
}

    handleButtonClick(index) {
        console.log('handle button clicked', index)
        this.setState({ currentSectionIndex: index})

    }

    renderContent(index) {
        const currentSection = this.props.sections[this.state.currentSectionIndex]
        console.log('Rendering content here')
        return (
          <p className='content'>
            { this.state.currentSectionIndex === index && currentSection.content}
          </p>
        )
      }



    renderButtons() {
        return this.props.sections.map((section, index) => (
            <li >
                   <button key={index} onClick={()=> this.handleButtonClick(index)}>{section.title}
                        <div>
                       {this.renderContent(index)}
                       </div>
                    </button>
                   
           </li>
       ))
    }

    render() {
    
    return(
        <ul>
            {this.renderButtons()}
        </ul>
    )
}
}
export default Accordion 