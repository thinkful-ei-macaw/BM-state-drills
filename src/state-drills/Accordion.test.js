import React from 'react';
import ReactDOM from 'react-dom';
import renderer from 'react-test-renderer'
import Accordion from './Accordion';
import { shallow } from 'enzyme'
import toJson from 'enzyme-to-json'

const sections = [
    {
      title: 'Section 1',
      content: 'Lorem ipsum dolor sit amet consectetur adipisicing elit.',
    },
    {
      title: 'Section 2',
      content: 'Cupiditate tenetur aliquam necessitatibus id distinctio quas nihil ipsam nisi modi!',
    },
    {
      title: 'Section 3',
      content: 'Animi amet cumque sint cupiditate officia ab voluptatibus libero optio et?',
    },
  ]



describe(`Accordion Component`, () => {
  it('opens any clicked question', () => {
    const wrapper = shallow(<Accordion sections={sections} />)
    wrapper.find('button').at(1).simulate('click')
    expect(toJson(wrapper)).toMatchSnapshot()
  })

  it('only opens one section at a time', () => {
    const wrapper = shallow(<Accordion sections={sections} />)
    wrapper.find('button').at(1).simulate('click')
    wrapper.find('button').at(2).simulate('click')
    expect(toJson(wrapper)).toMatchSnapshot()
  })

  
  it('renders no sections by default', () => {
    const tree = renderer.create(<Accordion sections={sections} />).toJSON()
    expect(tree).toMatchSnapshot()
  })

  it('renders empty given no Accordion', () => {
    const wrapper = shallow(<Accordion/>)
    expect(toJson(wrapper)).toMatchSnapshot()
})


})