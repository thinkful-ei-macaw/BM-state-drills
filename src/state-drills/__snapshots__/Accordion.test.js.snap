// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Accordion Component only opens one section at a time 1`] = `
<ul>
  <li>
    <button
      key="0"
      onClick={[Function]}
    >
      Section 1
      <div>
        <p
          className="content"
        />
      </div>
    </button>
  </li>
  <li>
    <button
      key="1"
      onClick={[Function]}
    >
      Section 2
      <div>
        <p
          className="content"
        />
      </div>
    </button>
  </li>
  <li>
    <button
      key="2"
      onClick={[Function]}
    >
      Section 3
      <div>
        <p
          className="content"
        >
          Animi amet cumque sint cupiditate officia ab voluptatibus libero optio et?
        </p>
      </div>
    </button>
  </li>
</ul>
`;

exports[`Accordion Component opens any clicked question 1`] = `
<ul>
  <li>
    <button
      key="0"
      onClick={[Function]}
    >
      Section 1
      <div>
        <p
          className="content"
        />
      </div>
    </button>
  </li>
  <li>
    <button
      key="1"
      onClick={[Function]}
    >
      Section 2
      <div>
        <p
          className="content"
        >
          Cupiditate tenetur aliquam necessitatibus id distinctio quas nihil ipsam nisi modi!
        </p>
      </div>
    </button>
  </li>
  <li>
    <button
      key="2"
      onClick={[Function]}
    >
      Section 3
      <div>
        <p
          className="content"
        />
      </div>
    </button>
  </li>
</ul>
`;

exports[`Accordion Component renders empty given no Accordion 1`] = `<ul />`;

exports[`Accordion Component renders no sections by default 1`] = `
<ul>
  <li>
    <button
      onClick={[Function]}
    >
      Section 1
      <div>
        <p
          className="content"
        />
      </div>
    </button>
  </li>
  <li>
    <button
      onClick={[Function]}
    >
      Section 2
      <div>
        <p
          className="content"
        />
      </div>
    </button>
  </li>
  <li>
    <button
      onClick={[Function]}
    >
      Section 3
      <div>
        <p
          className="content"
        />
      </div>
    </button>
  </li>
</ul>
`;

exports[`Accordion Component renders the first tab by default 1`] = `
<ul>
  <li>
    <button
      onClick={[Function]}
    >
      Section 1
      <div>
        <p
          className="content"
        />
      </div>
    </button>
  </li>
  <li>
    <button
      onClick={[Function]}
    >
      Section 2
      <div>
        <p
          className="content"
        />
      </div>
    </button>
  </li>
  <li>
    <button
      onClick={[Function]}
    >
      Section 3
      <div>
        <p
          className="content"
        />
      </div>
    </button>
  </li>
</ul>
`;
